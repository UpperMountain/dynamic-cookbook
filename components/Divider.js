import React from "react";
import { View, StyleSheet, Text } from "react-native";

const styles = StyleSheet.create({
  outer: {
    width: "100%",
    paddingLeft: "20%",
    paddingRight: "20%",
    flexDirection: "row",
    alignItems: "center"
  },
  line: {
    flex: 1,
    height: 1,
    backgroundColor: "#fff"
  },
  text: {
    textAlign: "center",
    flex: 0.3,
    color: "#fff"
  },
  container: {
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-evenly"
  }
});

export const HR = ({ text }) => {
  return (
    <View style={styles.outer}>
      <View style={[styles.line]} />
      <Text style={styles.text}>{text}</Text>
      <View style={[styles.line]} />
    </View>
  );
};

export const SpacedView = ({ height, children }) => {
  return <View style={[{ height }, styles.container]}>{children}</View>;
};
