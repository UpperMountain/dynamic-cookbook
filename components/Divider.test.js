import React from "react";
import renderer from "react-test-renderer";
import { HR, SpacedView } from "./Divider";

describe("SpacedView", () => {
  it("renders without exploding", () => {
    renderer.create(<SpacedView>Testing</SpacedView>);
  });

  it("passes snapshot tests", () => {
    const cmp = renderer.create(<SpacedView>Testing</SpacedView>);
    expect(cmp.toJSON()).toMatchSnapshot();
  });
});

describe("HR", () => {
  it("renders without exploding", () => {
    renderer.create(
      <SpacedView>
        Test1
        <HR />
        Test2
      </SpacedView>
    );
  });

  it("passes snapshot tests", () => {
    const cmp = renderer.create(
      <SpacedView>
        Test1
        <HR />
        Test2
        <HR />
        Test3
      </SpacedView>
    );
    expect(cmp.toJSON()).toMatchSnapshot();
  });
});
