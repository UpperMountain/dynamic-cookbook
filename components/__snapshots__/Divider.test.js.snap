// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HR passes snapshot tests 1`] = `
<View
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      },
    ]
  }
>
  Test1
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": "20%",
        "paddingRight": "20%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1,
          },
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "flex": 0.3,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1,
          },
        ]
      }
    />
  </View>
  Test2
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingLeft": "20%",
        "paddingRight": "20%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1,
          },
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "flex": 0.3,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1,
          },
        ]
      }
    />
  </View>
  Test3
</View>
`;

exports[`SpacedView passes snapshot tests 1`] = `
<View
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "space-evenly",
      },
    ]
  }
>
  Testing
</View>
`;
